simulator/risc_v/riscv_decoder.h

```diff
@@ -108,8 +178,15 @@ struct RISCVInstrDecoder
     explicit constexpr RISCVInstrDecoder(uint32 raw) noexcept
         : sz        ( apply_mask( raw, 0b00000000'00000000'00000000'00000011))
         , rd        ( apply_mask( raw, 0b00000000'00000000'00001111'10000000))
-        , rs1       ( apply_mask( raw, 0b00000000'00001111'10000000'00000000))
-        , rs2       ( apply_mask( raw, 0b00000001'11110000'00000000'00000000))
+        , rs1       ( apply_mask( raw, 0b11111111'11111111'00000000'00000000)     // If instr is 16-bit,
```

Comment:
That should use another variable instead of adding `?:` which creates great confusion here.